src/bp/ui-admin/src/telemetry.ts

```diff
@@ -112,72 +113,62 @@ export function checkInfoReceived() {
 }
 
-export function getServerFeedback() {
-  const pkgStr = window.localStorage.getItem('packageToSend')
-  let packages: Array<object> = []
-  if (pkgStr !== null) {
-    packages = JSON.parse(pkgStr)
+export async function feedbacks(feedbacks) {
+  try {
+    await axios.post(serverUrl, feedbacks, corsConfig)
+  } catch (err) {
+    console.log(err)
   }
-  return packages
 }
 
-export function feedback(pkg) {
-  axios
-    .post(serverUrl, pkg, corsConfig)
-    .then(res => {
-      const packages = getServerFeedback()
-      if (packages.indexOf(pkg) !== -1) {
-        packages.splice(packages.indexOf(pkg), 1)
-        window.localStorage.setItem('packageToSend', JSON.stringify(packages))
-      }
-      console.log(res)
-    })
-    .catch(err => {
-      const packages = getServerFeedback()
-      if (packages.indexOf(pkg) === -1) {
-        packages.push(pkg)
-        window.localStorage.setItem('packageToSend', JSON.stringify(packages))
-      }
-      console.log(err)
-    })
+export interface FeedbackType {
+  uuid: string
 }
 
-export function sendServerPackage() {
-  if (window.localStorage.getItem('packageToSend') === null) {
-    window.localStorage.setItem('packageToSend', JSON.stringify([]))
-  }
+export interface Feedback {
+  OK: Array<FeedbackType | null>
+  INACCESSIBLE: Array<FeedbackType | null>
+}
 
-  const packages = getServerFeedback()
-  packages.forEach((value, index) => {
-    feedback(value)
-  })
+export async function sendServerPackage() {
+  const packages = await axios.get(serverUrl, corsConfig)
 
-  axios
-    .get(serverUrl, corsConfig)
-    .then(res => {
-      if (_.has(res, 'data')) {
-        const payload = res.data.payload
-        const url = res.data.url
-        axios
-          .post(url, payload, corsConfig)
-          .then(res => {
-            feedback({ status: 'OK', data: payload })
-          })
-          .catch(err => {
-            feedback({ status: 'INACCESSIBLE', data: payload })
-            console.log(err)
-          })
+  let payload, url, feedbackUUID
+  const listFeedbacks: Feedback = {
+    OK: [],
+    INACCESSIBLE: []
+  }
+
+  if (packages !== undefined && _.has(packages, 'data')) {
+    for (const pkg of packages.data) {
+      if (pkg != null) {
+        payload = pkg.payload
+        payload.source = 'client'
+        url = pkg.url
+        feedbackUUID = payload.uuid
+
+        try {
+          await axios.post(url, payload, corsConfig)
+          listFeedbacks.OK.push({ uuid: feedbackUUID })
+        } catch (err) {
+          listFeedbacks.INACCESSIBLE.push({ uuid: feedbackUUID })
+          console.log(err)
+        }
       }
-    })
-    .catch(err => {
-      console.log(err)
-    })
+    }
+  }
+
+  await feedbacks(listFeedbacks)
 }
 
 export function setupServerPackageLoop() {
-  sendServerPackage()
+  sendServerPackage().catch(err => {
+    console.log(err)
+  })
   setInterval(() => {
-    sendServerPackage()
+    sendServerPackage().catch(err => {
```

Comment:
You can use `async` functions in `setInterval` just fine:
```
setInterval(async () => {
   try {
       await sendServerPackage()
   } catch (err) {
       console.log(err)
   }
}, ...)
```

Reply:
@spg are you sure about this ? I have a strange memory that setInterval are not good with async funcs

Reply:
yes it does, just tested it